File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int m_w = 0;
    int m_h = 0;
    for(vector<int> p : sizes) {
        int w = max(p[0], p[1]);
        int h = min(p[0], p[1]);
        
        m_w = max(m_w, w);
        m_h = max(m_h, h);
    }
    return m_w * m_h;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.swift
import Foundation

func solution(_ sizes:[[Int]]) -> Int {
    var maxW = 0, maxH = 0
    for size in sizes {
        let w = max(size[0], size[1])
        let h = min(size[0], size[1])
        maxW = max(maxW, w); maxH = max(maxH, h)
    }
    return maxW * maxH
}